resources/github-actions.md

```diff
@@ -45,6 +45,59 @@ There is a link provided at the end of every C++ and Python build job where the
+To do this, it is necessary to download CI artifacts (described in the above section) from one library during the CI workflow of another library.
+First, determine the pull request number and commit hash that need to be tested from the other library.
+Then, fetch the CI artifacts from the other library's pull request and use them when building and testing.
+The example code below demonstrates building and testing with conda packages from other library PRs.
+Replace the pull request numbers, commit hashes, and library names as needed.
+
+**Example 1:** Building `libcuml` (C++) using `librmm`, `libraft`, `libcumlprims_mg` PR artifacts. Adapted from [cuML PR 5318](https://github.com/rapidsai/cuml/pull/5318).
+
+```sh
+# ci/build_cpp.sh
+
+RAPIDS_CUDA_MAJOR="${RAPIDS_CUDA_VERSION%%.*}"
+LIBRMM_CHANNEL=$(rapids-get-artifact ci/rmm/pull-request/1223/72e0c74/rmm_conda_cpp_cuda${RAPIDS_CUDA_MAJOR}_$(arch).tar.gz)
+LIBRAFT_CHANNEL=$(rapids-get-artifact ci/raft/pull-request/1388/7bddaee/raft_conda_cpp_cuda${RAPIDS_CUDA_MAJOR}_$(arch).tar.gz)
+LIBCUMLPRIMS_CHANNEL=$(rapids-get-artifact ci/cumlprims_mg/pull-request/129/85effb7/cumlprims_mg_conda_cpp_cuda${RAPIDS_CUDA_MAJOR}_$(arch).tar.gz)
+
+# Build library packages with the CI artifact channels providing the updated dependencies
+
+rapids-mamba-retry mambabuild \
+    --channel "${LIBRMM_CHANNEL}" \
+    --channel "${LIBRAFT_CHANNEL}" \
+    --channel "${LIBCUMLPRIMS_CHANNEL}" \
+    conda/recipes/libcuml
+```
+
+**Example 2:** Testing `cudf` (Python) using `librmm`, `rmm`, and `libkvikio` PR artifacts. Adapted from [cuDF PR 12922](https://github.com/rapidsai/cudf/pull/12922).
+
+```sh
+# ci/test_python_common.sh
+
+RAPIDS_CUDA_MAJOR="${RAPIDS_CUDA_VERSION%%.*}"
+PYTHON_MINOR_VERSION=$(python --version | sed -E 's/Python [0-9]+\.([0-9]+)\.[0-9]+/\1/g')
+LIBRMM_CHANNEL=$(rapids-get-artifact ci/rmm/pull-request/1223/72e0c74/rmm_conda_cpp_cuda${RAPIDS_CUDA_MAJOR}_$(arch).tar.gz)
+RMM_CHANNEL=$(rapids-get-artifact ci/rmm/pull-request/1223/72e0c74/rmm_conda_python_cuda${RAPIDS_CUDA_MAJOR}_3${PYTHON_MINOR_VERSION}_$(arch).tar.gz)
+LIBKVIKIO_CHANNEL=$(rapids-get-artifact ci/kvikio/pull-request/224/68febbb/kvikio_conda_cpp_cuda${RAPIDS_CUDA_MAJOR}_$(arch).tar.gz)
+
+# Install library packages with the CI artifact channels providing the updated dependencies for testing
+
+rapids-mamba-retry install \
+  --channel "${CPP_CHANNEL}" \
+  --channel "${PYTHON_CHANNEL}" \
+  --channel "${LIBRMM_CHANNEL}" \
+  --channel "${LIBKVIKIO_CHANNEL}" \
+  --channel "${RMM_CHANNEL}" \
+  cudf libcudf
+```
+
+Note that the custom channel for PR artifacts is needed in the build scripts _and_ the test scripts.
+If building/testing a Python package that depends on a C++ library, it is also necessary to use PR artifacts from that C++ library (e.g. if testing `rmm` artifacts, use the corresponding `librmm` CI artifacts as well).
+
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**💡 suggestion: (non-blocking)** ‏ You could add a section with the tip you gave me to automatically grab the commit hash given the PR: 

```
REPO="rmm"
PR_NUMBER="1095"
COMMIT=$(git ls-remote https://github.com/rapidsai/${REPO}.git refs/heads/pull-request/${PR_NUMBER} | cut -c1-7)
```

Reply:
I'll rewrite this with https://github.com/rapidsai/gha-tools/pull/85.